src/erlcloud_cloudformation.erl

```diff
@@ -0,0 +1,412 @@
+            StackSummaries = extract_stack_summaries(xmerl_xpath:string("/ListStacksResponse/ListStacksResult", XmlNode)),
+            {ok, StackSummaries, NextToken};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec list_stack_resources_all(string(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+list_stack_resources_all(Params, Config = #aws_config{}) ->
+    list_all(fun list_stack_resources/2, Params, Config, []).
+
+-spec list_stack_resources(params(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+list_stack_resources(Params, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "ListStackResources", Params) of
+        {ok, XmlNode} ->
+            NextToken = erlcloud_xml:get_text("/ListStackResourcesResponse/ListStackResourcesResult/NextToken", XmlNode, undefined),
+            StackResources = extract_list_stack_resources(xmerl_xpath:string("/ListStackResourcesResponse/ListStackResourcesResult", XmlNode)),
+            {ok, StackResources, NextToken};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec describe_stack_resources(string(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+describe_stack_resources(Params, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "DescribeStackResources", Params) of
+        {ok, XmlNode} ->
+            extract_stack_resources_members(xmerl_xpath:string("/DescribeStackResourcesResponse", XmlNode));
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec describe_stack_resource(params(), aws_config) -> {ok, cloudformation_list()} | {error, _}.
+describe_stack_resource(Param, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "DescribeStackResource", Param) of
+        {ok, XmlNode} ->
+            extract_stack_details(xmerl_xpath:string("/DescribeStackResourceResponse", XmlNode));
+        {error, Error} ->
+            {error, Error}
+    end.
+
+
+-spec describe_stacks_all(params(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+describe_stacks_all(Params, Config = #aws_config{}) ->
+    list_all(fun describe_stacks/2, Params, Config, []).
+
+-spec describe_stacks(params(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+describe_stacks(Params, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "DescribeStacks", Params) of
+        {ok, XmlNode} ->
+            NextToken = erlcloud_xml:get_text("/DescribeStacksResponse/DescribeStacksResult/NextToken", XmlNode, undefined),
+            DescribedStacks = extract_described_stacks_result(xmerl_xpath:string("/DescribeStacksResponse", XmlNode)),
+            {ok, DescribedStacks, NextToken};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec get_stack_policy(params(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+get_stack_policy(Params, Config = #aws_config{}) ->
```

Comment:
@lilrooness please re-check that 
1) every required parameter is the one of functions parameter
2) every not required parameter is being mapped from lower_case_with_underscores notation to "CamelCasesWithoutUnderscorers" notation